frontend/lib/parse-capi/index.ts

```diff
@@ -82,11 +96,11 @@ const findPillar: (name: string) => Pillar | undefined = name => {
 };
 
 const getLink = (data: {}, { isPillar }: { isPillar: boolean }): LinkType => ({
-    title: getNonEmptyString(data, 'title'),
-    longTitle: getString(data, 'longTitle'),
-    url: getNonEmptyString(data, 'url'),
-    pillar: isPillar ? findPillar(getNonEmptyString(data, 'title')) : undefined,
-    children: getArray(data, 'children').map(
+    title: getString(data, 'title', ''),
+    longTitle: getString(data, 'longTitle', ''),
+    url: getString(data, 'url', ''),
+    pillar: isPillar ? findPillar(getString(data, 'title', '')) : undefined,
```

Comment:
I think `title` and `longTitle` should still be non-empty strings. The reason some pages are crashing (I think) is because the [subnav expects a `parent` to be defined](https://github.com/guardian/dotcom-rendering/blob/master/frontend/lib/parse-capi/index.ts#L200), but `parent` ought to be optional. 

Consider:

![screen shot 2018-09-24 at 17 37 00](https://user-images.githubusercontent.com/5931528/45965493-87861d00-c020-11e8-8505-134d1021212c.png)

vs

![screen shot 2018-09-24 at 17 37 20](https://user-images.githubusercontent.com/5931528/45965501-8bb23a80-c020-11e8-8c57-e124762db24c.png)

Perhaps we should only add a `parent` property to `subNavSections` if `subnav.parent` is defined

Reply:
Ah nice spot, thanks. I spent a long time looking through the config json trying to work out why title might be null, this makes more sense!

Reply:
Yeah, it's really non-obvious. We should make clearer documentation or more obvious types around these config objects.

We should also make [`NavType.subNavSections.parent`](https://github.com/guardian/dotcom-rendering/blob/master/frontend/index.d.ts#L29) [optional](https://www.typescriptlang.org/docs/handbook/interfaces.html#optional-properties) for clarity.

Let me know if you need any help with any of this 🙂